libraries/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts

```diff
@@ -845,10 +845,14 @@ export class PhasedScriptAction extends BaseScriptAction<IPhasedCommandConfig> {
           }
 
           const { startTime, endTime } = operationResult.stopwatch;
+
           operationResults[operation.name!] = {
             startTimestampMs: startTime,
             endTimestampMs: endTime,
             nonCachedDurationMs: operationResult.nonCachedDurationMs,
+            wasExecutedOnThisMachine:
+              !operationResult.cobuildRunnerId ||
+              operationResult.cobuildRunnerId === cobuildConfiguration?.cobuildRunnerId,
```

Comment:
Strictly speaking, does a replay from cache count as "executed on this machine"?

Reply:
Open to other names here. Strictly speaking, kind of/kind of not right? Cobuilds are intended to be state restores across machines as though it was built on any given cobuild agent, but from the data, that build wasn't executed on this machine b/c the cobuild runner ids from the state file and machine's cobuild config don't match.

Reply:
@dmichon-msft Sorry lost track of this, would also be open to just adding the cobuildRunnerId to the metadata object and leaving it up to the user to make this check.

Reply:
Or alternatively, we could leave as is and go forward with #4680, but advise folks to only send telemetry from a single agent.